File: android/src/main/kotlin/com/affise/attribution/affise_attribution_lib/ResultWrapper.kt
package com.affise.attribution.affise_attribution_lib

import com.affise.attribution.internal.callback.InternalResult
import io.flutter.plugin.common.MethodChannel

class ResultWrapper(private val methodResult: MethodChannel.Result) : InternalResult {

    override fun success(data: Any?) {
        methodResult.success(data)
    }

    override fun error(error: String) {
        methodResult.error(error, null, null)
    }

    override fun notImplemented() {
        methodResult.notImplemented()
    }
}

File: ios/Classes/ResultWrapper.swift
import Flutter
import AffiseInternal

class ResultWrapper : InternalResult {
    
    private var methodResult: FlutterResult
    
    public init(_ result: @escaping FlutterResult) {
        self.methodResult = result
    }

    func success(_ result: Any?) {
        methodResult(result)
    }
    
    public func error(_ error: String) {
        methodResult(FlutterError.init(
            code: error,
            message: error,
            details: nil
        ))
    }
        
    public func notImplemented() {
        methodResult(FlutterMethodNotImplemented)
    }
}
